### ISSUE TITLE
Switch button smaller and included inside border

### ISSUE THREAD

**Issue by clobrano** [ubuntu/gtk-communitheme#44]
Like in settings, make the switches (green/grey) smaller and so more elegant. If possible, shrink the inner toggle also a little to not touch the border of the switch.

**Comment by godlyranchdressing** [comment]
![7f5af1b5851e47e12029833fb90d470d8a83c66e](https://user-images.githubusercontent.com/27529229/34919997-0808820c-f942-11e7-8e8a-435b42582c3c.gif)

(Borders are done with box-shadow so as to not add pxs to the switch)

**Comment by clobrano** [comment]
@godlyranchdressing are we working on the same bug?

**Comment by godlyranchdressing** [comment]
@clobrano nope, just copy and pasting the mockup and notes to here. Though I did close issue #55 (which was due to border-color being used instead of overwriting the last box-shadow)

**Comment by clobrano** [comment]
Awesome, thanks!

**Comment by madsrh** [comment]
I still think the switches look huge. What do you guys think, can I reopen?

Here compared to Arc:
![toggels](https://user-images.githubusercontent.com/3986894/36279828-ae776030-1298-11e8-86a1-915842f8d2d8.gif)

**Comment by clobrano** [comment]
I had hard time trying to reduce the size more than that, not sure why, but let me try again

**Comment by galgalesh** [comment]
I actually like them the way they are now. I don't mind big buttons, I think it's part of the Suru look; much like the folder icons and the font, which look a bit bloated too. I think it makes it look a bit more friendly.)

**Comment by Luxamman** [comment]
Suru was very much for mobile and touch input, I also would like see if they can be a little smaller (at least in height) to look less clumsy. 
Let's see what @clobrano can do :)

**Comment by clobrano** [comment]
Currently I am stuck with this

![image](https://user-images.githubusercontent.com/2883614/36814390-a1e0464a-1cd7-11e8-8aac-cd4279ae07c6.png)

I couldn't style headerbar switch, no idea why, and couldn't make knob square :thinking:

**Comment by ya-d** [comment]
I played around with switches as well to adopt the new check/radio button style and mix that with that (now obsolete) button design. I can't currently think about that any further or update the design to reflect the new direction for buttons, so I simply post it as a thought.

* "Improvements": white knob with borders as radio/check buttons; background reflects button design (what it used to be); white knob in backdrop windows indicates active state
* Versions: left is current design – middle is proposed design – right adds border to the background as have buttons and pretty much anything else but button overlays.
* As for size we could at least try to match the size of the knob with checkboxes and/or volume sliders and see how that looks. (I didn't mean to change size for the mockups.)

![switch button](https://user-images.githubusercontent.com/30781876/36836343-545ca776-1d39-11e8-922b-682dfafe2624.png)

**Comment by nana-4** [comment]
@clobrano 

> couldn't style headerbar switch

It needs to increase top/bottom `margin` for `headerbar switch`.
```css
headerbar switch {
  margin-top: 10px;
  margin-bottom: 9px;
}
```

> couldn't make knob square

Try `switch { font-size: 0; }` :smile:

---

However, IMHO 15px knob looks too small.
![image](https://user-images.githubusercontent.com/9556384/36831123-bc153af6-1d69-11e8-9cc1-2591f7128d06.png)
I'd prefer 20px knob (which is same size with the current scale slider)
![image](https://user-images.githubusercontent.com/9556384/36831135-c6747e44-1d69-11e8-9f1f-ffbd78b082bc.png)
...or 22px knob (which is only 1px smaller than the current knob).
![image](https://user-images.githubusercontent.com/9556384/36831127-c017d8b6-1d69-11e8-8fd4-032022373b53.png)

**Comment by ya-d** [comment]
+1 for 20px on first sight.

**Comment by clobrano** [comment]
I agree with @nana-4, 15px doesn't look too bad, but it's weird

Rounded maybe?
![image](https://user-images.githubusercontent.com/2883614/37719082-8222df7c-2d24-11e8-8dd3-349e998ad640.png)

squared and 18px looks this way
![image](https://user-images.githubusercontent.com/2883614/37719183-b9e35784-2d24-11e8-8035-85f1dbff85c1.png)

**Comment by nana-4** [comment]
IMO the rounded one will lose consistency with the GtkScale's slider. I personally like the current roundness (even if we round the GtkScale's slider or not).

(Sorry for nitpicking, but the knobs of the bottom picture seems to be actually 20px. I measured it. :) )

**Comment by nana-4** [comment]
@clobrano It's because there is an invisible 1px border. So that need to be 16px in the code. :)

**Comment by zappedfish** [comment]
It would be nice if squares inside were smaller. They are too big right now and create too much contrast because of green background that also looks like a green border outside of a square

**Comment by Feichtmeier** [comment]
@clobrano I am somehow unsatisfied with the togglebuttons, too :|
They could either use a small grey border or any other trick I don't know. They look... unfinnished (to me! :) )

**Comment by gamunu** [comment]
Better if we can have material like toggle buttons. 
![image](https://user-images.githubusercontent.com/4501687/39395703-5eb0cb6e-4aff-11e8-8ad1-b966e99ca47b.png)
This is from PopOS theme.

**Comment by Feichtmeier** [comment]
@gamunu I disagree, while they look good on material inspired themes like materia/pop/adapta, they won't look fitting for this theme. This is not very close to material design. It's mostly flat yet, but it's different. I don't think our just needs some refinement when we have time for this.

**Comment by clobrano** [comment]
Trying again some mockups

![image](https://user-images.githubusercontent.com/2883614/39531459-7a3fce64-4e2b-11e8-8496-1523811ce16c.png)

**Comment by ya-d** [comment]
I really like the 3d-effect! 👍

But it looks slightly different to the one for (pressed) buttons, doesn't it? Would be great if you could match those. Also I still think matching the design of sliders, check boxes and radio buttons by adding a grey border is a good idea (see my earlier [proposal](https://github.com/ubuntu/gtk-communitheme/issues/44#issuecomment-369527569)). Maybe a border to the "background" is also what's missing to make it more like the button style.

**Comment by Feichtmeier** [comment]
Much better! I agree a super light grey 1px border could increase the desire to toggle them other than that I'd prefer those to the current ones

**Comment by clobrano** [comment]
I was aiming at replicating not pressed buttons actually. I'll try again with your suggestions

**Comment by clobrano** [comment]
New iteration

![image](https://user-images.githubusercontent.com/2883614/39571421-baad4aa8-4ecb-11e8-83b0-076fccec7414.png)

**Comment by ya-d** [comment]
> I think that fits to the "3d depth" of pressed buttons! I like it

Me too! 👍 Should the off background be slightly lighter (225 instead of 205 rgb) to match pressed buttons? (205 is used by inactive elements like check boxes and stuff.) Just a thought.

**Comment by clobrano** [comment]
I think unchecked switches should not match pressed buttons colors, since they represent opposite states (even if different widgets), maybe unpressed buttons?

**Comment by clobrano** [comment]
![image](https://user-images.githubusercontent.com/2883614/39576524-b5e5afd4-4ede-11e8-9208-d875d0fe71b1.png)

**Comment by ya-d** [comment]
I like both versions but lean a bit towards the brighter one. Thanks for trying 😉

What bothers me more is how users recognize it as "off" compared to "disabled". The depth already helps with that, maybe even more if you **lift** the knob by the same effect. Borders might be enough though.

**Comment by Feichtmeier** [comment]
The knob could indeed receive a super soft border, like the progress sliders in gnome music

**Comment by clobrano** [comment]
I am a bit torn here. I added a border to the knob and increased the box-shadow to look more like the one in the switch. It looks good to me when unselected, but the selected one puzzles me

![image](https://user-images.githubusercontent.com/2883614/39620782-96f5f30a-4f8c-11e8-8e8f-4a9d5494a41d.png)

I have tried first using a darker success_color, but it creates some glitches at the round corners, so here I am using the very same colors as the unselected switch. Maybe too many colors (green, grey, white) in the same little spot?

**Comment by madsrh** [comment]
IMO there's too much shadow here. I liked the previous one better.

**Comment by clobrano** [comment]
Too much shadow on the knob, or also on the switch?

**Comment by Feichtmeier** [comment]
Yes was an idea and you tested it. The previous one looked definitely better

**Comment by madsrh** [comment]
The knob 🎛 

Sorry, I should let @Feichtmeier and @ya-d comment, because I was very happy with the version 23 hours ago 😆

**Comment by clobrano** [comment]
:smile: so we have a winner I think https://github.com/ubuntu/gtk-communitheme/issues/44#issuecomment-386249751

**Comment by Feichtmeier** [comment]
Yes the super 3d one looks almost like it would fall of the slider like a piece of paper upon a board that has not enough glue xD
